floss/language/go/extract.py

```diff
@@ -0,0 +1,431 @@
+                            addr = m.start() + pe.OPTIONAL_HEADER.ImageBase + section_va
+                            yield StaticString(string=data, offset=addr, encoding=StringEncoding.ASCII)
+
+                    except UnicodeDecodeError:
+                        continue
+
+        if section_name == ".text":
+            # Extract string in .text section
+            section_va = section.VirtualAddress
+            section_size = section.SizeOfRawData
+            section_data = section.get_data(section_va, section_size)
+
+            for m in combinedregex.finditer(section_data):
+                for i in range(1, 8):
+                    try:
+                        tmp_string = m.group(i)
+                        if tmp_string != b"":
+                            try:
+                                decoded_string = tmp_string.decode("utf-8")
+                                if decoded_string.isprintable() and len(decoded_string) >= min_length:
+                                    addr = m.start() + pe.OPTIONAL_HEADER.ImageBase + section_va
+                                    yield StaticString(
+                                        string=decoded_string, offset=addr, encoding=StringEncoding.ASCII
+                                    )
+                            except UnicodeDecodeError:
+                                pass
+                    except AttributeError:
+                        pass
+
+        if section_name == ".rdata":
+            # Extract string blob in .rdata section that starts with "go:buidid" or "go.buildid"
+            """
+            67 6F 3A 62 75 69 6C 64  69 64 00 69 6E 74 65 72  go:buildid.inter
+            6E 61 6C 2F 63 70 75 2E  49 6E 69 74 69 61 6C 69  nal/cpu.Initiali
+            7A 65 00 69 6E 74 65 72  6E 61 6C 2F 63 70 75 2E  ze.internal/cpu.
+            70 72 6F 63 65 73 73 4F  70 74 69 6F 6E 73 00 69  processOptions.i
+            6E 74 65 72 6E 61 6C 2F  63 70 75 2E 69 6E 64 65  nternal/cpu.inde
+            78 42 79 74 65 00 69 6E  74 65 72 6E 61 6C 2F 63  xByte.internal/c
+            """
+            section_va = section.VirtualAddress
+            section_size = section.SizeOfRawData
+            section_data = section.get_data(section_va, section_size)
+
+            blob_pattern = re.compile(b"\x67\x6F(\x2E|\x3A)\x62\x75\x69\x6C\x64\x69\x64\x00(.)*\x00\x00", re.DOTALL)
+            for m in blob_pattern.finditer(section_data):
+                t = m.group(0)
+                for s in t.split(b"\x00"):
+                    try:
+                        x = s.decode("utf-8")
+                        if x.isprintable() and x != "" and len(x) >= min_length:
+                            addr = m.start() + pe.OPTIONAL_HEADER.ImageBase + section_va
+                            yield StaticString(string=x, offset=addr, encoding=StringEncoding.ASCII)
+                    except UnicodeDecodeError:
+                        pass
+
+        if section_name == ".idata":
```

Comment:
are these the names from the import table?

Reply:
Yes, these names are indeed extracted from the import table. Is there any issue regarding the extraction of string data from the import table?

Reply:
we could do it more elegantly using pefile (potentially supporting other section names), low prio though